2200044/문제1.py

```diff
@@ -0,0 +1,28 @@
+for j in range(10):
+    stack = []
+    k = int(input())
+    l = input()
+    for i in range(k):
+        if l[i] == '(':
+            stack.append('(')
+        if l[i] == '[':
+            stack.append('[')
+        if l[i] == '{':
+            stack.append('{')
+        if l[i] == '<':
+            stack.append('<')
+        if i > 0:
+            if l[i] == '>' and stack[len(stack)-1] == '<':
+                stack.pop()
+            if l[i] == ']' and stack[len(stack)-1] == '[':
+                stack.pop()
+            if l[i] == '}' and stack[len(stack)-1] == '{':
+                stack.pop()
+            if l[i] == ')' and stack[len(stack)-1] == '(':
+                stack.pop()
+        else:
+            break
```

Comment:
```suggestion
    for i in range(k):
        if l[i] == '(':
            stack.append('(')
        elif l[i] == '[':
            stack.append('[')
        elif l[i] == '{':
            stack.append('{')
        elif l[i] == '<':
            stack.append('<')
        elif i > 0:
            if l[i] == '>' and stack[len(stack)-1] == '<':
                stack.pop()
            elif l[i] == ']' and stack[len(stack)-1] == '[':
                stack.pop()
            elif l[i] == '}' and stack[len(stack)-1] == '{':
                stack.pop()
            elif l[i] == ')' and stack[len(stack)-1] == '(':
                stack.pop()
            else:
                break
        else:
            break
```
지금 코드는 **i가 0일 때 바로 탐색 종료**하도록 구현되어 있어요! `elif`를 활용하시면 이 문제를 해결할 수 있습니다.